src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const sortByCentury = century
+    ? people.filter((person) => Math.ceil(person.died / 100) === century)
+    : people;
+
+  const sortByGender = sortByCentury.filter((person) => person.sex === 'm');
+
+  const averageAge = sortByGender.reduce((acc, person) => acc
+    + (person.died - person.born), 0)
+      / sortByGender.length;
```

Comment:
on my opinion, its better to separate this part of code by several variables, where you will find every person's age first, and then find average age
its improve readability of code

Reply:
as for me my var has own name and inside her name we can see about what she is

Reply:
but ok, i'll add your advice